language/src/regent/codegen.t

```diff
@@ -4461,6 +4537,70 @@ function codegen.expr_list_range(cx, node)
     expr_type)
 end
 
+function codegen.expr_list_ispace(cx, node)
+  local ispace_type = std.as_read(node.ispace.expr_type)
+  local ispace = codegen.expr(cx, node.ispace):read(cx, ispace_type)
+  local expr_type = std.as_read(node.expr_type)
+
+  local result = terralib.newsymbol(expr_type, "result") -- Resulting list.
+
+  -- Construct AST that populates the `result` list with indices from `ispace`.
+  --
+  -- Loop variable: `for p in ispace do`
+  local p_symbol = std.newsymbol(
+    -- TODO(zhangwen): how necessary is it to give a "bound"?  is this how you do it?
+    ispace_type.index_type(std.newsymbol(ispace_type)),
```

Comment:
local bound
  if value:is(ast.typed.expr.ID) then
    bound = value.value
  else
    bound = std.newsymbol(value_type)
  end